src/lib/tests/task.test.ts

```diff
@@ -5,145 +5,239 @@ import type { SvelteConcurrencyUtils } from '$lib/task';
 import Default from './components/default.svelte';
+import Enqueue from './components/enqueue.svelte';
+import Drop from './components/drop.svelte';
 
 function wait(ms: number) {
 	return new Promise((resolve) => setTimeout(resolve, ms));
 }
 
-describe('task - default handler', () => {
-	it('calls the function you pass in', async () => {
-		const fn = vi.fn();
-		const { getByTestId } = render(Default, {
-			fn,
-		});
-		const perform = getByTestId('perform-default');
-		perform.click();
-		await vi.waitFor(() => {
-			expect(fn).toHaveBeenCalled();
-		});
-	});
+function all_options(fn: (selector: string) => void) {
+	describe.each(['default', 'options'])('version with %s', fn);
+}
 
-	it("runs to completion if it's not cancelled", async () => {
-		let count = 0;
-		const wait_time = 2000;
-		async function* fn() {
-			await wait(wait_time);
-			yield;
-			count++;
-		}
-		const { getByTestId } = render(Default, {
-			fn,
+describe.each([
+	{ component: Default, name: 'default' },
+	{ component: Enqueue, name: 'enqueue' },
+	{ component: Drop, name: 'drop' },
+])('task - basic functionality $name', ({ component }) => {
+	all_options((selector) => {
+		it('calls the function you pass in', async () => {
+			const fn = vi.fn();
+			const { getByTestId } = render(component, {
+				fn,
+			});
+			const perform = getByTestId(`perform-${selector}`);
+			perform.click();
+			await vi.waitFor(() => {
+				expect(fn).toHaveBeenCalled();
+			});
 		});
-		const perform = getByTestId('perform-default');
-		perform.click();
-		await vi.waitFor(
-			() => {
-				expect(count).toBe(1);
-			},
-			{
-				timeout: 3000,
-			},
-		);
-	});
 
-	it("doesn't runs to completion if it's cancelled, the function is a generator and there's a yield after every await", async () => {
-		let count = 0;
-		const wait_time = 2000;
-		let task_signal: AbortSignal;
-		async function* fn(_: number, { signal }: SvelteConcurrencyUtils) {
-			task_signal = signal;
-			await wait(wait_time);
-			yield;
-			count++;
-		}
-		const { getByTestId } = render(Default, {
-			fn,
+		it("runs to completion if it's not cancelled", async () => {
+			let count = 0;
+			const wait_time = 2000;
+			async function* fn() {
+				await wait(wait_time);
+				yield;
+				count++;
+			}
+			const { getByTestId } = render(component, {
+				fn,
+			});
+			const perform = getByTestId(`perform-${selector}`);
+			perform.click();
+			await vi.waitFor(
+				() => {
+					expect(count).toBe(1);
+				},
+				{
+					timeout: 3000,
+				},
+			);
 		});
-		const perform = getByTestId('perform-default');
-		const cancel = getByTestId('cancel-default');
-		perform.click();
-		await wait(500);
-		cancel.click();
-		await vi.waitFor(() => {
-			expect(task_signal.aborted).toBeTruthy();
+
+		it("doesn't runs to completion if it's cancelled, the function is a generator and there's a yield after every await", async () => {
+			let count = 0;
+			const wait_time = 2000;
```

Comment:
This makes the tests quite slow, could we reduce this to the maximum stable number?